### PR TITLE
Feature: (ISA-161) Query statistics for habitat

### PR DESCRIPTION
**Pull request by JoshGx2000**
[ISA-161](https://jira.its.utas.edu.au/browse/ISA-161)

We've got habitat statistics (from FINALPRODUCT_SeamapAus) for arbitrary networks/parks/zones combinations!

Screen Shot 2022-06-17 at 10 40 41 am

Screen Shot 2022-06-17 at 10 40 50 am

Screen Shot 2022-06-17 at 10 41 07 am

### CODE REVIEW

File: database/Views/SeamapAus_BoundaryAreas_View.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  SUM([area]) AS [area]
+FROM (
+  SELECT
+    [network],
+    [park],
+    [zone],
+    [zone_iucn],
+    MAX([boundary_area]) AS [area]
+  FROM [IMASSeamap].[dbo].[SeamapAus_HabitatStatistics_View]
+  GROUP BY [polygon_id], [network], [park], [zone], [zone_iucn]
```
[the remark this answers is not in the document]
Comment (JoshGx2000) [reply database/Views/SeamapAus_BoundaryAreas_View.sql]: I believe I had a reason yesterday (I just checked and removing it _does_ affect the output). I believe it had something to do with not counting duplicate polygons, but looking at it now I may have messed up my code (unsure). I've drastically simplified this view now, which is good

File: backend/habitat/viewsets.py [hunk 2]
```diff
@@ -518,30 +537,27 @@ def habitat_statistics(request):
...
+            ' and '.join(f"{v['type']}=\'{v['value']}\'" for v in boundaries if v['value']),
+            ', '.join(v['type'] for v in boundaries if v['value'])
+        )
+        cursor.execute(boundary_area_sql)
+        try:
+            boundary_area = float(cursor.fetchone()[0])
+
+            habitat_stats_sql = SQL_GET_HABITAT_STATS.format(
+                ' and '.join(f"{v['type']}=\'{v['value']}\'" for v in boundaries if v['value'])
+            )
```
[the remark this answers is not in the document]
Comment (JoshGx2000) [reply backend/habitat/viewsets.py:559]: Whoops, absolutely! I completely forgot about that. I'll fix it up now.

[the remark this answers is not in the document]
Comment (JoshGx2000) [reply backend/habitat/viewsets.py]: A Django model might be overkill? I took the note you left [here](https://github.com/IMASau/Seamap/blob/36efbe8acd1776edd395da358b8895b8131e626a/backend/habitat/viewsets.py#L29-L35) and applied it to this problem, so now we have [this](https://github.com/IMASau/Seamap/blob/e506797203b946f4610f3f7c5a0878c159f2f0e5/backend/habitat/viewsets.py#L548-L552) and [this](https://github.com/IMASau/Seamap/blob/e506797203b946f4610f3f7c5a0878c159f2f0e5/backend/habitat/viewsets.py#L556-L559).

**Comment by JoshGx2000** [comment]
Self-reminder that when this is merged and deployed to remove the old views that are going unused.

File: backend/habitat/viewsets.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (markhepburn) [reply backend/habitat/viewsets.py]: Yeah, I'll pay that, that should work.  Clarity is another question, possibly, but let's go with that for now.

**Pull request merged**